libdnf/dnf-repo.cpp

```diff
@@ -973,6 +973,25 @@ dnf_repo_set_keyfile_data(DnfRepo *repo, GError **error)
 
     /* baseurl is optional; if missing, unset it */
     baseurls = g_key_file_get_string_list(priv->keyfile, repoId, "baseurl", NULL, NULL);
+    if (baseurls) {
+        // baseruls can be ['value1', 'value2, value3'] therefore we first join to have 'value1, value2, value3'
+        tmp_strval = g_strjoinv(",", baseurls);
```

Comment:
What about don't use/remove "global" `tmp_strval`. Instead it can be defined at places where it is used.
So, you can write this: 
`g_autofree gchar * tmp_strval = g_strjoinv(",", baseurls);`
Everyone then knows that the value of the variable is valid/used only in this code block. 
Moreover, `g_autofree` will be working correctly and you can delete the line: `g_free(g_steal_pointer(&tmp_strval));`

Reply:
Updated, thanks.